src/MostUsed/CLI.hs

```diff
@@ -27,13 +28,28 @@ withInfo opts h d f =
     info (helper <*> opts) $ header h <> progDesc d <> footer f
 
 parseOptions :: Parser Options
-parseOptions = Options <$> parseIncludeFirstArgument <*> parseDebug
+parseOptions = Options
+    <$> parseIncludeFirstArgument
+    <*> parseDebug
+    <*> parseShell
 
 parseIncludeFirstArgument :: Parser [String]
 parseIncludeFirstArgument = many $ strOption $
     long "include-first-argument"
+    <> metavar "command_name"
     <> help "Count this command with its first argument (can be specified more than once)"
 
+parseShell :: Parser Shell
+parseShell = fmap shell $ strOption $
+    long "shell"
+    <> metavar "[bash | zsh]"
+    <> help "Which type of shell history to parse"
+
+shell :: String -> Shell
+shell "bash" = Bash
+shell "zsh" = Zsh
+shell _ = error "--shell can only take 'bash' or 'zsh'"
```

Comment:
Right, I'm sort of OK with throwing an error right as the option is parsed. Though the error is a little gross:

```
most-used-exe: --shell can only take 'bash' or 'zsh'
CallStack (from HasCallStack):
  error, called at src/MostUsed/CLI.hs:51:11 in most-used-0.1.0.0-5xJL0L1se38JcakkBKWPOD:MostUsed.CLI
```